File: src/net/auoeke/cin/element/BooleanElement.java
package net.auoeke.cin.element;

public class BooleanElement implements Element {
    public boolean value;

    public BooleanElement(boolean value) {
        this.value = value;
    }

    @Override public Type type() {
        return Element.Type.BOOLEAN;
    }

    @Override public boolean equals(Object other) {
        return other instanceof BooleanElement bool && this.value == bool.value;
    }

    @Override public int hashCode() {
        return Boolean.hashCode(this.value);
    }

    @Override public String toString() {
        return Boolean.toString(this.value);
    }
}

File: src/net/auoeke/cin/element/BooleanElement.kt
package net.auoeke.cin.element

class BooleanElement(var value: Boolean) : Element {
    override val type: ElementType get() = ElementType.BOOLEAN

    override fun equals(other: Any?): Boolean = other is BooleanElement && other.value == value
    override fun hashCode(): Int = value.hashCode()
    override fun toString(): String = value.toString()
}
